### PR TITLE
Display nicer URLs when editing a link

### PR DESCRIPTION
**Pull request by notnownikki**
Filters the URL for display by removing protocol, www subdomain, and if there path is '/', the trailing slash.

### CODE REVIEW

**Comment by notnownikki** [comment]
Fixes https://github.com/WordPress/gutenberg/issues/2174

File: blocks/editable/format-toolbar/index.js [hunk 1]
```diff
@@ -40,6 +41,19 @@ const FORMATTING_CONTROLS = [
 // Default controls shown if no `enabledControls` prop provided
 const DEFAULT_CONTROLS = [ 'bold', 'italic', 'strikethrough', 'link' ];
 
+// TODO: rename and move up one level, write tests
+export const filterURIForDisplay = ( uri ) => {
```
Comment (youknowriad) [inline blocks/editable/format-toolbar/index.js]: Should we remove this? :)

File: blocks/editable/format-toolbar/index.js [hunk 2]
```diff
@@ -182,7 +196,7 @@ class FormatToolbar extends Component {
 							href={ this.state.linkValue }
 							target="_blank"
 						>
-							{ this.state.linkValue && decodeURI( this.state.linkValue ) }
+							{ this.state.linkValue && filterURLForDisplay( decodeURI( this.state.linkValue ) ) }
```
Comment (youknowriad) [inline blocks/editable/format-toolbar/index.js]: I wonder if we should move the `decodeURI` call inside the `filterURLForDisplay` function

Comment (notnownikki) [inline blocks/editable/format-toolbar/index.js]: I almost did that, but I thought that perhaps not all the URLs we want to display "nicely" would begin encoded, so left it out.

File: editor/utils/url.js [hunk 3]
```diff
@@ -25,3 +25,22 @@ export function getGutenbergURL( query = {} ) {
...
+/**
+ * Returns a url for display
+ *
+ * @param  {String} url    Original url
+ *
+ * @return {String}        Displayed URL
+ */
+export function filterURLForDisplay( url ) {
+	// remove protocol and www prefixes
+	const filteredURL =	url.replace( new RegExp( '^https?://(www\.)?' ), '' );
```
Comment (mtias) [inline editor/utils/url.js]: There's an extra tab there :)

Comment (notnownikki) [inline editor/utils/url.js]: This is why spaces are superior ;) (fixed)

**Comment by notnownikki** [comment]
Rebased.

**Comment by mtias** [comment]
Looking good to me.

**Pull request merged**